From 14dc092d4bebe929b6a530cee7ee7141113436c5 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Tue, 26 Mar 2019 11:26:29 -0700
Subject: [PATCH 0580/1851] v4l: xilinx: vpss-scaler: Use 10bit media bus codes

commit f8ee6f3e1287cbe772bdb39dbc96e21e366a78b1 from
https://github.com/Xilinx/linux-xlnx.git

This patch enables driver to use new YUV 10bit media bus codes.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index 21612a33222b..e07d7eaad259 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -1198,21 +1198,25 @@ xv_vscaler_setup_video_fmt(struct xscaler_device *xscaler, u32 code_in)
 
 	switch (code_in) {
 	case MEDIA_BUS_FMT_VYYUYY8_1X24:
+	case MEDIA_BUS_FMT_VYYUYY10_4X20:
 		dev_dbg(xscaler->xvip.dev,
 			"Vscaler Input Media Format YUV 420");
 		video_in = XVIDC_CSF_YCRCB_420;
 		break;
 	case MEDIA_BUS_FMT_UYVY8_1X16:
+	case MEDIA_BUS_FMT_UYVY10_1X20:
 		dev_dbg(xscaler->xvip.dev,
 			"Vscaler Input Media Format YUV 422");
 		video_in = XVIDC_CSF_YCRCB_422;
 		break;
 	case MEDIA_BUS_FMT_VUY8_1X24:
+	case MEDIA_BUS_FMT_VUY10_1X30:
 		dev_dbg(xscaler->xvip.dev,
 			"Vscaler Input Media Format YUV 444");
 		video_in = XVIDC_CSF_YCRCB_444;
 		break;
 	case MEDIA_BUS_FMT_RBG888_1X24:
+	case MEDIA_BUS_FMT_RBG101010_1X30:
 		dev_dbg(xscaler->xvip.dev,
 			"Vscaler Input Media Format RGB");
 		video_in = XVIDC_CSF_RGB;
@@ -1265,21 +1269,25 @@ static int xv_hscaler_setup_video_fmt(struct xscaler_device *xscaler,
 
 	switch (code_out) {
 	case MEDIA_BUS_FMT_VYYUYY8_1X24:
+	case MEDIA_BUS_FMT_VYYUYY10_4X20:
 		dev_dbg(xscaler->xvip.dev,
 			"Hscaler Output Media Format YUV 420\n");
 		video_out = XVIDC_CSF_YCRCB_420;
 		break;
 	case MEDIA_BUS_FMT_UYVY8_1X16:
+	case MEDIA_BUS_FMT_UYVY10_1X20:
 		dev_dbg(xscaler->xvip.dev,
 			"Hscaler Output Media Format YUV 422\n");
 		video_out = XVIDC_CSF_YCRCB_422;
 		break;
 	case MEDIA_BUS_FMT_VUY8_1X24:
+	case MEDIA_BUS_FMT_VUY10_1X30:
 		dev_dbg(xscaler->xvip.dev,
 			"Hscaler Output Media Format YUV 444\n");
 		video_out = XVIDC_CSF_YCRCB_444;
 		break;
 	case MEDIA_BUS_FMT_RBG888_1X24:
+	case MEDIA_BUS_FMT_RBG101010_1X30:
 		dev_dbg(xscaler->xvip.dev,
 			"Hscaler Output Media Format RGB\n");
 		video_out = XVIDC_CSF_RGB;
@@ -1501,6 +1509,7 @@ static int xscaler_set_format(struct v4l2_subdev *subdev,
 				XSCALER_MIN_WIDTH, xscaler->max_pixels);
 	format->height = clamp_t(unsigned int, fmt->format.height,
 				 XSCALER_MIN_HEIGHT, xscaler->max_lines);
+	format->code = fmt->format.code;
 	fmt->format = *format;
 	return 0;
 }
-- 
2.31.1

